src/compiler/utils/builder.js

```diff
@@ -59,5 +59,32 @@ export default {
 
   elTypes: {
     escapableRaw: ['textarea', 'title']
+  },
+
+  /**
+   * whether we are in an html_attribute given the current state
+   * NOTE: when we see a tornado body previous states are reset.
```

Comment:
I like the idea of optimizing. I suppose having an html-like string in a `data-` attribute isn't actually malformed, so we should probably support this instead of throw an error.